srv/modules/runners/ui-iscsi.py

```diff
@@ -0,0 +1,233 @@
+
+log = logging.getLogger(__name__)
+
+
+class Iscsi(object):
+    """
+    Populating the view requires network interfaces by host, images by pool
+    and the current lrbd configuration.
+    Saving the choices requires writing out the lrbd.conf
+    """
+
+    def __init__(self):
+	"""
+	Set yaml dumper, initialize Salt client
+	"""
+	self.data = {}
+	self.friendly_dumper = yaml.SafeDumper
+	self.friendly_dumper.ignore_aliases = lambda self, data: True
+	self.local = salt.client.LocalClient()
+
+    def populate(self):
+	"""
+	Return the lrbd configuration, interfaces and images.
+	"""
+	self.data['config'] = self.config()
+	self.data['interfaces'] = self.interfaces()
+	self.data['images'] = self.images()
+
+	return self.data
+
+    def interfaces(self, wrapped=True):
+	"""
+	Parse grains for all network interfaces on igw roles.  Possibly
+	select public interface.
+	"""
+	_stdout = sys.stdout
+	sys.stdout = open(os.devnull, 'w')
+
+	igws = self.local.cmd("I@roles:igw", 'grains.get', ['ipv4'], expr_form="compound")
+	sys.stdout = _stdout
+	if wrapped:
+	    data = []
+	    for igw in igws.keys():
+		for addr in igws[igw]:
+		    if addr == '127.0.0.1':
+			continue
+		    data.append({'node': igw, 'addr': addr})
+	    return data
+	else:
+	    for igw in igws.keys():
+		igws[igw].remove('127.0.0.1')
+
+	    return igws
+
+    def images(self, wrapped=True):
+	"""
+	Read a Salt mine of cephimages that lists all images
+	and their pools.
+	"""
+	__opts__ = salt.config.client_config('/etc/salt/master')
+	result = salt.utils.minions.mine_get('I@roles:master',
+					     'cephimages.list',
+					     'compound', __opts__)
+	if wrapped:
+	    data = []
+	    for master in result.keys():
+		for pool in result[master]:
+		    data.append({'pool': pool, 'img': result[master][pool]})
+		break
+	    return data
+	else:
+	    for master in result.keys():
+		return result[master]
+
+    def config(self, filename="/srv/salt/ceph/igw/cache/lrbd.conf"):
+	"""
+	Read the existing lrbd.conf
+	"""
+	if os.path.exists(filename):
+	    return json.loads(open(filename).read())
+
+    def save(self, filename="/srv/salt/ceph/igw/cache/lrbd.conf", **kwargs):
+	"""
+	Convert data to lrbd sections if necessary.
+	Ensure ceph.igw.config is disabled (or better create sls file
+	that says configured by gui)
+	Overwrite /srv/salt/ceph/igw/cache/lrbd.conf
+	"""
+	if 'data' in kwargs:
+	    self._set_igw_config(**kwargs)
+	    contents = kwargs['data']
+	    with open(filename, 'w') as conf:
+		conf.write(json.dumps(contents, indent=4) + '\n')
+	else:
+	    log.error("No JSON data passed")
+
+    def _set_igw_config(self, cluster='ceph', **kwargs):
+	"""
+	Add igw_config to cluster.yml
+
+	Comments are removed from cluster.yml
+	"""
+	if 'filename' in kwargs:
+	    filename = kwargs['filename']
+	else:
+	    filename = '/srv/pillar/ceph/stack/{}/cluster.yml'.format(cluster)
+	contents = {}
+	with open(filename, 'r') as yml:
+	    contents = yaml.safe_load(yml)
```

Comment:
If the file `filename` is empty or just contains comment lines then variable `contents` will have the value `None` and will throw an exception in the next line.
Please add the following to the next line:
```
if not contains:
    contents = {}